conda_libmamba_solver/index.py

```diff
@@ -227,9 +230,9 @@ def _fetch_channel(self, url: str) -> tuple[str, os.PathLike]:
         if "PYTEST_CURRENT_TEST" in os.environ:
             # Workaround some testing issues - TODO: REMOVE
             # Fix conda.testing.helpers._patch_for_local_exports by removing last line
-            maybe_cached = SubdirData._cache_.get((url, self._repodata_fn))
-            if maybe_cached and maybe_cached._mtime == float("inf"):
-                del SubdirData._cache_[(url, self._repodata_fn)]
+            for key, cached in list(SubdirData._cache_.items()):
+                if key[:2] == (url, self._repodata_fn) and cached._mtime == float("inf"):
```

Comment:
All the SubdirData cache keys used to be a two-item tuple. Now it's three because we include `add_pip_as_python_dependency` too. If we change the key system, it would break in way more ways too, but if you want we can get defensive and check whether `key` is a tuple first.

Reply:
Yeah, can't hurt, can it?

Reply:
Added a check.

Reply:
Thank you!